tiledb/sm/query/reader.h

```diff
@@ -363,6 +385,67 @@ class Reader {
       uint64_t* buffer_size,
       bool check_null_buffers = true);
 
+  /**
+   * Sets the data for a fixed/var-sized attribute/dimension.
+   *
+   * @param name The attribute/dimension to set the buffer for.
+   * @param buffer The buffer that will hold the data to be read.
+   * @param buffer_size This initially contains the allocated
+   *     size of `buffer`, but after the termination of the function
+   *     it will contain the size of the useful (read) data in `buffer`.
+   * @param check_null_buffers If true (default), null buffers are not
+   * allowed.
+   * @return Status
+   */
+  Status set_buffer_data(
+      const std::string& name,
+      void* buffer,
+      uint64_t* buffer_size,
+      bool check_null_buffers = true);
+
+  /**
+   * Sets the offset buffer for a var-sized attribute/dimension.
+   *
+   * @param name The attribute/dimension to set the buffer for.
+   * @param buffer_off The buffer that will hold the data to be read.
+   *     This buffer holds the starting offsets of each cell value in
+   *     `buffer_val`.
+   * @param buffer_off_size This initially contains
+   *     the allocated size of `buffer_off`, but after the termination of the
+   *     function it will contain the size of the useful (read) data in
+   *     `buffer_off`.
+   * @param check_null_buffers If true (default), null buffers are not
+   * allowed.
+   * @return Status
+   */
+  Status set_buffer(
```

Comment:
same here, use `_offsets` and `_validity` as postfixes on API names that they are related to.